src/client/scripts/game/chess/legalmoves.js

```diff
@@ -237,6 +240,17 @@ const legalmoves = (function(){
             }
         }
 
+        for (var line in legalMoves.slides) {
+            line=line.split(',') // can't hash array so conversion to string made, unconverting it
+            let limits = legalMoves.slides[line];
```

Comment:
I think as this line is, it's a length-2 array of strings. I think to access properties from `slides` like this it has to call line.toString() first, which I believe is slow. Can `${line[0]},${line[1]}` be used instead?

Reply:
Correction, after further reviewing. Calling .toString() on an array is 100-fold SLOWER than using a template literal or concatenation.. yes ONE HUNDRED fold 😅 Me and tsevasa were optimizing the ICN notation converter.

Reply:
I believe all occurrences of this have been changed to use the `math.js` methods. If you find one, please point it out